### ISSUE TITLE
[Security] Moderate Severity Vulnerability in `axios` Affects `apify-client`

### ISSUE THREAD

**Issue by mustpax** [apify/apify-client-js#501]
## Description

A recent `npm audit` has identified a moderate severity vulnerability in the `axios` package, specifically related to Cross-Site Request Forgery (CSRF), which affects versions 0.8.1 to 1.5.1 of `axios`. The `apify-client` package, which depends on `axios`, is therefore impacted by this vulnerability. This issue has been documented in the GitHub security advisories [GHSA-wf5p-g6vw-rhxx](https://github.com/advisories/GHSA-wf5p-g6vw-rhxx).

## Affected Version

The vulnerability affects the following versions of `apify-client` that depend on the vulnerable versions of `axios`:

- apify-client versions >=0.10.0

## Impact

The vulnerability allows attackers to perform CSRF attacks on web applications, which could compromise the security of applications using `apify-client` and indirectly, the vulnerable versions of `axios`.

## Steps to Reproduce

1. Run `npm audit` in a project that depends on `apify-client`.
2. Notice the reported moderate severity vulnerability related to `axios`.

## Expected Behavior

Dependencies of `apify-client` should not have known moderate or high severity vulnerabilities that can impact the security of its users.

## Possible Solution

It is recommended to update the `axios` dependency within `apify-client` to the latest version that has addressed this CSRF vulnerability. If there are breaking changes associated with the update, a major version bump for `apify-client` may be necessary to avoid impacting existing users negatively.

## Additional Context

Given the widespread use of `axios` in the JavaScript ecosystem, addressing this vulnerability promptly is crucial to ensuring the security of applications using `apify-client`. We understand that dependency management can be challenging, especially regarding potential breaking changes, but we appreciate your efforts to maintain the security and reliability of your package.

Thank you for your attention to this matter and your continued support of the open-source community.

**Comment by mtrunkat** [comment]
@B4nan could somebody from tooling take a look?

**Comment by B4nan** [comment]
I already looked into this several times during last year, but since there is no migration guide for axios v1 (which is a huge ignorance on their end, how come?), it's far away from simple. I'd say it would be better if someone who originally put this together could look into it, as it's not clear what the use of axios is about (it's pretty complex and uses removed/undocumented APIs).

I'll have another look today to refresh my mind on this, but I failed multiple times when trying to do the upgrade while making the test suite green.
